.github/workflows/cd.yml

```diff
@@ -0,0 +1,64 @@
+name: cd
+on:
+  workflow_run:
+    workflows: ["ci"]
+    types:
+      - completed
+  push:
+    tags:
+      - "v[0-9]+.[0-9]+.[0-9]+"
+
+permissions:
+  contents: read
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@dc323e67f16fb5f7663d20ff7941f27f5809e9b6 # v3.2.0
+        with:
+          fetch-depth: 0
+
+      - name: cd/build-docker
+        run: make build-image
+
+      - name: ci/build-docker-e2e
+        run: make build-image-e2e
+
+      - name: ci/scan-docker-security
+        uses: aquasecurity/trivy-action@9ab158e8597f3b310480b9a69402b419bc03dbd5 # v0.8.0
+        with:
+          image-ref: "mattermost/mattermost-cloud"
+          format: "table"
+          exit-code: "1"
+          ignore-unfixed: true
+          vuln-type: "os,library"
+          severity: "CRITICAL"
+          output: "trivy-results-critical.sarif"
+
+      - name: ci/scan-docker-security-e2e
+        uses: aquasecurity/trivy-action@9ab158e8597f3b310480b9a69402b419bc03dbd5 # v0.8.0
+        with:
+          image-ref: "mattermost/mattermost-cloud-e2e"
+          format: "table"
+          exit-code: "1"
+          ignore-unfixed: true
+          vuln-type: "os,library"
+          severity: "CRITICAL"
+          output: "trivy-results-critical-e2e.sarif"
+
+      - name: Upload trivy scan Provisioner
+        uses: github/codeql-action/upload-sarif@896079047b4bb059ba6f150a5d87d47dde99e6e5 # v2.11.6
+        with:
+          sarif_file: "trivy-results-critical.sarif"
+
+      - name: Upload trivy scan E2E
+        uses: github/codeql-action/upload-sarif@896079047b4bb059ba6f150a5d87d47dde99e6e5 # v2.11.6
+        with:
+          sarif_file: "trivy-results-critical-e2e.sarif"
+
+      - name: cd/docker-push
+        run: ./scripts/push-docker.sh
+        env:
+          TAG: ${GITHUB_REF_NAME}
```

Comment:
Suggest moving the push functionality to a Makefile, and drop all related bash scripts laying around.

Example; 
https://github.com/mattermost/rtcd/blob/master/Makefile#L151-L163

Reply:
Resolving since it is out of the scope. We are only migrating to ghactions in this iteration. 
Makefile concerns to be addressed in next iteration.